sqlglot/tokens.py

```diff
@@ -626,13 +638,15 @@ def tokenize(self, sql):
                 break
 
             white_space = self.WHITE_SPACE.get(self._char)
-            identifier_end = self.IDENTIFIERS.get(self._char)
+            identifier_end = self._IDENTIFIERS.get(self._char)
 
             if white_space:
                 if white_space == TokenType.BREAK:
                     self._col = 1
                     self._line += 1
-            elif self._char == "0" and self._peek == "x":
+            elif self._char == "0" and self._peek.upper() == "B":
+                self._scan_bits()
+            elif self._char == "0" and self._peek.upper() == "X":
```

Comment:
Dialects that support 0x and 0X nomenclature need to be processed here, so the leading digit doesn't confuse things. Later, formats like X'FF' are processed via the trie. This is a bit annoying because there are two hex processing methods now. Do you think that's okay?

Reply:
Should this logic be checking that 0x or 0X is supported by the dialect before trying to process as a hex number? There are a lot of hardcoded symbols in the scanning logic and I wasn't sure if I should follow that norm or try to make the code smarter (but more complex).

Reply:
can we avoid doing self._char == 0 and self._peek.upper twice? could we maybe put it inside the self._char.isdigit() block?

so like

elif self.char.isdigit():
   if self._char == "0":
      peek = self._peek.upper()
...

Reply:
refactored